File: templates/Features/SuspendAndResume/Services/OnBackgroundEnteringEventArgs.cs
using System;

namespace Param_ItemNamespace.Services
{
    public class OnBackgroundEnteringEventArgs : EventArgs
    {
        public SuspensionState SuspensionState { get; set; }

        public Type Target { get; private set; }

        public OnBackgroundEnteringEventArgs(SuspensionState suspensionState, Type target)
        {
            SuspensionState = suspensionState;
            Target = target;
        }
    }
}

File: templates/Uwp/Features/SuspendAndResumeVB/Services/OnBackgroundEnteringEventArgs.vb
Namespace Services
    Public Class OnBackgroundEnteringEventArgs
        Inherits EventArgs

        Public Property SuspensionState As SuspensionState
        
        Public Property Target As Type

        Public Sub New(_suspensionState As SuspensionState, _target As Type)
            SuspensionState = _suspensionState
            Target = _target
        End Sub
      End Class
End Namespace
